### PR TITLE
Cleanup docstrings for `gammapy.utils`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This addresses part of issue https://github.com/gammapy/gammapy/issues/4811, specifically for `gammapy.utils`

### CODE REVIEW

**Comment by registerrier** [comment]
@Astro-Kirsty . This should be rebased.

File: gammapy/utils/array.py [hunk 1]
```diff
@@ -25,15 +25,15 @@ def array_stats_str(x, label=""):
 
     Parameters
     ----------
-    x : array-like
-        Array
+    x : `~numpy.ndarray`
```
Comment (registerrier) [inline gammapy/utils/array.py:28]: I would keep array-like here

File: gammapy/utils/metadata.py [hunk 2]
```diff
@@ -191,6 +191,29 @@ def validate_time(cls, v):
         else:
             return v
 
+    def to_header(self, format="gadf"):
```
Comment (registerrier) [inline gammapy/utils/metadata.py:194]: This should not be in this PR. There might have been a problem when rebasing.

File: gammapy/utils/testing.py [hunk 3]
```diff
@@ -159,6 +159,8 @@ def assert_time_allclose(actual, desired, atol=1e-3):
 def assert_quantity_allclose(actual, desired, rtol=1.0e-7, atol=None, **kwargs):
     """Assert all-close for `astropy.units.Quantity` objects.
```
Comment (registerrier) [inline gammapy/utils/testing.py:160]:
```suggestion
    """Assert all-close for `~astropy.units.Quantity` objects.
```

File: gammapy/utils/time.py [hunk 4]
```diff
@@ -34,19 +34,20 @@ def time_to_fits(time, epoch=None, unit=u.s):
     Parameters
     ----------
     time : `~astropy.time.Time`
-        time to be converted
-    epoch : `astropy.time.Time`
-        epoch to use for the time. The corresponding keywords must
+        Time to be converted.
+    epoch : `astropy.time.Time`, optional
```
Comment (registerrier) [inline gammapy/utils/time.py:38]:
```suggestion
    epoch : `~astropy.time.Time`, optional
```

File: gammapy/utils/time.py [hunk 5]
```diff
@@ -34,19 +34,20 @@ def time_to_fits(time, epoch=None, unit=u.s):
...
+        Epoch to use for the time. The corresponding keywords must
         be stored in the same FITS header.
-        If None, `DEFAULT_EPOCH` will be used.
-    unit : `astropy.time.Time`
+        Default is None, so the `DEFAULT_EPOCH` is used.
+    unit : `astropy.time.Time`, optional
```
Comment (registerrier) [inline gammapy/utils/time.py:42]:
```suggestion
    unit : `~astropy.units.Unit`, optional
```

File: gammapy/utils/time.py [hunk 6]
```diff
@@ -34,19 +34,20 @@ def time_to_fits(time, epoch=None, unit=u.s):
...
         If None, `DEFAULT_EPOCH` will be used.
```
Comment (registerrier) [inline gammapy/utils/time.py:43]: This should be removed here, no?

File: gammapy/utils/time.py [hunk 7]
```diff
@@ -62,19 +63,20 @@ def time_to_fits_header(time, epoch=None, unit=u.s):
     Parameters
     ----------
     time : `~astropy.time.Time`
-        time to be converted
-    epoch : `astropy.time.Time`
-        epoch to use for the time. The corresponding keywords must
+        Time to be converted.
+    epoch : `astropy.time.Time`, optional
```
Comment (registerrier) [inline gammapy/utils/time.py:67]:
```suggestion
    epoch : `~astropy.time.Time`, optional
```

File: gammapy/utils/time.py [hunk 8]
```diff
@@ -62,19 +63,20 @@ def time_to_fits_header(time, epoch=None, unit=u.s):
...
+        Epoch to use for the time. The corresponding keywords must
         be stored in the same FITS header.
-        If None, `DEFAULT_EPOCH` will be used.
-    unit : `astropy.time.Time`
+        Default is None, so `DEFAULT_EPOCH` is used.
+    unit : `astropy.time.Time`, optional
```
Comment (registerrier) [inline gammapy/utils/time.py:71]:
```suggestion
    unit : `~astropy.units.Unit`, optional
```

File: gammapy/utils/time.py [hunk 9]
```diff
@@ -62,19 +63,20 @@ def time_to_fits_header(time, epoch=None, unit=u.s):
...
         If None, `DEFAULT_EPOCH` will be used.
```
Comment (registerrier) [inline gammapy/utils/time.py:72]: remove here.

**Review by registerrier — DISMISSED** [review]
Thanks @Astro-Kirsty . See small inline comments.

Can you check why there is a `to_header` method that was added. Was there an issue in the rebase?

File: gammapy/utils/metadata.py (hunk 2 above)
Comment (registerrier) [reply gammapy/utils/metadata.py:194]: Well, it has been removed with #4886 . You are introducing a former version of it...

Comment (Astro-Kirsty) [reply gammapy/utils/metadata.py:194]: Ah apologies, I was looking at the wrong part of the code. I will rebase again

Comment (Astro-Kirsty) [reply gammapy/utils/metadata.py]: Had to fix it manually as it was a commit that I had made, should be all good now

File: gammapy/utils/coordinates/other.py [hunk 10]
```diff
@@ -73,12 +73,12 @@ def velocity_glon_glat(x, y, z, vx, vy, vz):
 
 def motion_since_birth(v, age, theta, phi):
     """
-    Compute motion of a object with given velocity, direction and age.
+    Compute motion of an object with given velocity, direction and age.
```
Comment (MRegeard) [inline gammapy/utils/coordinates/other.py:76]: `object` kind of clashes with python `object`. We could use `source` or `astrophysics object`?

File: gammapy/utils/docs.py [hunk 11]
```diff
@@ -36,9 +36,7 @@
 
 
 class AccordionHeader(Directive):
-    """
-    Inserts HTML code to open an accordion box in the How To.
-    """
+    """Inserts HTML code to open an accordion box in the How To."""
```
Comment (MRegeard) [inline gammapy/utils/docs.py:39]: pydocstyle ask for imperative wording, so it should be like this:

```suggestion
    """Insert HTML code to open an accordion box in the How To."""
```

File: gammapy/utils/docs.py [hunk 12]
```diff
@@ -83,9 +81,7 @@ def run(self):
 
 
 class AccordionFooter(Directive):
-    """
-    Inserts HTML code to close an accordion box in the How To.
-    """
+    """Inserts HTML code to close an accordion box in the How To."""
```
Comment (MRegeard) [inline gammapy/utils/docs.py:84]: Same here ?

File: gammapy/utils/parallel.py [hunk 13]
```diff
@@ -9,14 +9,14 @@
 
 
 class ParallelBackendEnum(Enum):
-    """Enum for parallel backend"""
+    """Enum for parallel backend."""
```
Comment (MRegeard) [inline gammapy/utils/parallel.py:12]: Here `Enum` is a keyword for `Enumeration`, so do we want to put the full name or keep the keyword ?

**Review by MRegeard — COMMENTED** [review]
Thanks @Astro-Kirsty, see inline comments.

File: gammapy/utils/parallel.py (hunk 13 above)
Comment (Astro-Kirsty) [reply gammapy/utils/parallel.py:12]: As the docstring is so short I thought keeping it as the keyword is best

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty .

**Pull request merged**